core/src/main/scala/org/apache/spark/shuffle/ShuffleBlockResolver.scala

```diff
@@ -34,7 +34,7 @@ trait ShuffleBlockResolver {
    * Retrieve the data for the specified block. If the data for that block is not available,
    * throws an unspecified exception.
    */
-  def getBlockData(blockId: ShuffleBlockId): ManagedBuffer
+  def getBlockData(shuffleGenerationId: Int, blockId: ShuffleBlockId): ManagedBuffer
```

Comment:
seems better to make it `def getBlockData(shuffleId: Int, shuffleGenerationId: Int, mapId...)` if there are not many existing callers.

Reply:
Thanks for the advice, done in 7259500.